elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -160,19 +166,28 @@ protected Integer calculateRecordsGRAPHQL(String response, String table_name) {
      * @return the table name from the above query
      */
     protected String getTableNameFromQuery(String jsonStr) {
-        JSONObject j = new JSONObject(jsonStr);
-        String s = (String) j.get("query");
-        String str = "";
-        for (int i = 0; i < s.length(); i++) {
-            if (s.charAt(i) == '{' || s.charAt(i) == '}') {
-                continue;
-            }
-            else {
-                str = str + s.charAt(i);
+        StringBuilder str = new StringBuilder();
+        String[] queryTerms;
+        try {
+            JSONObject j = new JSONObject(jsonStr);
+            String s = (String) j.get("query");
+
+            Integer countBrackets = 0;
+            for (int i = 0; i < s.length(); i++) {
+                if (s.charAt(i) == '{' || s.charAt(i) == '}') {
+                    countBrackets++;
+                    if (countBrackets == 2) {
+                        break;
+                    }
+                }
+                else {
+                    str.append(s.charAt(i));
+                }
             }
+            queryTerms = str.toString().trim().split("\\s+");
+        } catch (JSONException e) {
+            queryTerms = str.toString().trim().split("\\s+");
```

Comment:
instead of storing the value in `queryTerms', you can just do log.error. Remove line 187 and 170. And in line 191 do `return str.toString().trim().split("\\s+");`

Reply:
made the above changes and returned str.toString().trim().split("\s+")[0] instead of str.toString().trim().split("\s+") as we need to return a String.